lib/std/unicode.zig

```diff
@@ -196,22 +196,119 @@ pub fn utf8CountCodepoints(s: []const u8) !usize {
     return len;
 }
 
-pub fn utf8ValidateSlice(s: []const u8) bool {
+// Ported from go, which is licensed under a BSD-3 license.
+// https://golang.org/LICENSE
+//
+// https://golang.org/src/unicode/utf8/utf8.go
+/// Returns true if the input consists entirely of UTF-8 runes
+pub fn utf8ValidateSlice(input: []const u8) bool {
+    var p = input;
+
+    // Fast path. Check for and skip 8 bytes of ASCII characters per iteration.
+    while (p.len >= 8) {
+        const first32 = @as(u32, p[0]) | @as(u32, p[1]) << 8 | @as(u32, p[2]) << 16 | @as(u32, p[3]) << 24;
+        const second32 = @as(u32, p[4]) | @as(u32, p[5]) << 8 | @as(u32, p[6]) << 16 | @as(u32, p[7]) << 24;
+        if ((first32 | second32) & 0x80808080 != 0) {
+            // Found a non ASCII byte
+            break;
+        }
+        p = p[8..];
+    }
+
+    const RuneSelf = 0x80;
```

Comment:
Zig does not have a `Rune` type like Go, thus this should renamed to something like `rune_self`.